src/DynamoApplications/AssemblyConflictChecker.cs

```diff
@@ -0,0 +1,134 @@
+﻿using System;
+using System.IO;
+using System.Collections.Generic;
+using System.Linq;
+using System.Reflection;
+using DynamoApplications.Properties;
+
+namespace Dynamo.Applications
+{
+    /// <summary>
+    /// Checks for incompatibilities between loaded assemblies.
+    /// </summary>
+    public class AssemblyConflictChecker : IDisposable
```

Comment:
This class is designed to make it easier for the API user to log assembly conflict notifications